src/include/storage/sql_table.h

```diff
@@ -170,10 +177,37 @@ class SqlTable {
     return {initializer, projection_map};
   }
 
+  /**
+   * This is really hacky, just for test purposes only. Because the current transaction and logging infrastructure are
+   * all based on data tables, but checkpointing is implemented on sql tables. Migrating existing API from data tables
+   * to sql tables would require much effort that would need to modify many tests, and this is left for future work.
+   * The current workaround is to get the underlying data table and do not modify existing infrastructure.
+   * @return the data table underlying sqltable
+   */
+  storage::DataTable *get_data_table() { return table_.data_table; }
+
+  /**
+   * This is extremely hacky, just for test purposes only. Because the current transaction and logging infrastructure
+   * are all based on data tables, but checkpointing is implemented on sql tables. Migrating existing API from data
+   * tables to sql tables would require much effort that would need to modify many tests, and this is left for future
+   * work. The current workaround is to get the underlying data table and do not modify existing infrastructure.
+   * @return the data table underlying sqltable
+   */
+  storage::BlockLayout get_layout() { return table_.layout; }
```

Comment:
Since existing logging and transaction infrastructure uses data table in APIs, this hack here is required to test our existing code. Because we have to get a data_table and layout for logging purposes.
Migrating logging and transaction infrastructure to SqlTable seems out of our project's scope (involving many transaction layer APIs as well) and we think is dangerous because it breaks many tests. So we think there should better be a separate PR addressing that in the future. 
In this sense this PR would indeed not be "merge ready" before the infrastructure has changed, but we are wondering if the workaround is acceptable here and be fixed in the future.

Reply:
From my point of view, it is not a hack to make our code pass CI checks. It is only a hack that enables us to use the existing test infrastructure to test our code. Since the existing test infrastructure is still based on DataTable and Layout, we need some hacky way to get the Layout to use it. However, our code is still get tested and it does not change the result of the test itself. We never use that API in our implementation code.